File: anylang/go/leetcode/1512.好数对的数目.go
package leetcode

import "fmt"

func getNums(x int) int {
	return x * (x - 1) / 2
}

func numIdenticalPairs(nums []int) int {
	var rst = 0
	countMap := make(map[int]int)

	for _, v := range nums {
		count := countMap[v]
		if count > 0 {
			countMap[v] = count + 1
		} else {
			countMap[v] = 1
		}
	}

	for _, v := range countMap {
		rst += getNums(v)
	}

	return rst
}

func NumIdenticalPairs() {
	arr := []int{1, 2, 3, 1, 1, 3}
	fmt.Println("numIdenticalPairs([1,2,3,1,1,3])", numIdenticalPairs(arr))
}

File: anylang/c++/1512.好数对的数目.cpp
#include <stdio.h>
#include <iostream>
#include <map>
#include <vector>

class Solution {
   public:
    int numIdenticalPairs(std::vector<int>& nums) {
        int rst = 0;
        std::map<int, int> hash;

        for (int i = 0; i < nums.size(); i++) {
            // int val = nums[i];
            // int count = hash[val];
            // hash[val] = count > 0 ? ++count : 1;

            std::cout << hash[nums[i]] << std::endl;

            ++hash[nums[i]];
        }

        for (const auto& [key, val] : hash) {
            rst += getNums(val);
        }

        return rst;
    }

   private:
    int getNums(int x) {
        return x * (x - 1) / 2;
    }
};

int main() {
    std::vector<int> arr = {1, 2, 3, 1, 1, 3};
    Solution s;

    printf("numIdenticalPairs({1,2,3,1,1,3}) %d", s.numIdenticalPairs(arr));
    return 0;
}